# Patch from Andy Hauser 24feb05
--- pico.h	2005-02-19 18:45:11.000000000 +0100
+++ pico.h	2005-02-19 18:44:32.000000000 +0100
@@ -12,7 +12,7 @@
 #include <math.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <dirent.h>
-#include <termio.h>
+#include <termios.h>
 #include <setjmp.h>
 #include <signal.h>
 #include <dlfcn.h>
--- net.c	2005-02-19 18:49:01.000000000 +0100
+++ net.c	2005-02-19 18:53:45.000000000 +0100
@@ -8,6 +8,7 @@
 #include <sys/socket.h>
 #include <arpa/inet.h>
 #include <netinet/tcp.h>
+#include <netinet/in.h>
 
 static void tcpErr(any ex, char *s) {
    err(ex, NULL, "TCP %s error: %s", s, strerror(errno));
--- Makefile	2005-02-19 19:27:29.000000000 +0100
+++ Makefile	2005-02-19 19:27:51.000000000 +0100
@@ -27,7 +27,7 @@
 $(bin)/pico: $(picoFiles:.c=.o)
 	mkdir -p $(bin) $(lib) ../tmp ../fifo
 	echo "  " link pico:
-	gcc  -rdynamic  -o $(bin)/pico  $(picoFiles:.c=.o)  -lc -lm -ldl
+	gcc  -dynamic  -o $(bin)/pico  $(picoFiles:.c=.o)  -lc -lm
 	strip $(bin)/pico
 
 $(lib)/ext: ext.o
